### PR TITLE
Project structured, basic app layout, CI fixes

### PR DESCRIPTION
**Pull request by S-Maciejewski**
This PR reflects changes done by the dev team until the beginning of October. The changes include mainly fundamentals of app layout and proper project structure.

Issues: #3 #43

### CODE REVIEW

File: .github/workflows/dart.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ name: Dart CI
 on:
   push:
     branches:
-      - master
+      - dev-release-test_fixed
```
Comment (szymon-owczarzak) [inline .github/workflows/dart.yml:6]: This line probably needs to be reverted

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # cogboardmobileapp
 
-Cogboard mobile application project
+Mobile CogBoard app
```
Comment (szymon-owczarzak) [inline README.md:3]: I preferer `Cogboard Mobile` here

File: lib/models/material_colors.dart [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  50: Color.fromRGBO(150, 150, 150, .1),
+  100: Color.fromRGBO(150, 150, 150, .2),
+  200: Color.fromRGBO(150, 150, 150, .3),
+  300: Color.fromRGBO(150, 150, 150, .4),
+  400: Color.fromRGBO(150, 150, 150, .5),
+  500: Color.fromRGBO(150, 150, 150, .6),
+  600: Color.fromRGBO(150, 150, 150, .7),
+  700: Color.fromRGBO(150, 150, 150, .8),
+  800: Color.fromRGBO(150, 150, 150, .9),
+  900: Color.fromRGBO(150, 150, 150, 1),
```
Comment (szymon-owczarzak) [inline lib/models/material_colors.dart:16]: I would think that when using `MaterialColor` - we would already have a whole palette without the need to define all of this.

**Review by szymon-owczarzak — COMMENTED** [review]
I'm not that fluent in dart so let's go through the code during tomorrow's meeting.

File: lib/screens/dashboards_screen.dart [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        selectedItemColor: dashboardTabs[dashboardsProvider.dashboardTabIndex]
+            .selectedTabColor,
+        unselectedItemColor: Theme.of(context).accentColor,
+        currentIndex: dashboardsProvider.dashboardTabIndex,
+        items: [
+          BottomNavigationBarItem(
+            backgroundColor: Theme.of(context).primaryColor,
+            icon: Icon(Icons.star),
+            title: Text('Favourite'),
+          ),
```
Comment (szymon-owczarzak) [inline lib/screens/dashboards_screen.dart:39]: extract to a method maybe?

File: lib/screens/dashboards_screen.dart [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        onTap: (index) => dashboardsProvider.setDashboardTabIndex(index),
+        selectedItemColor: dashboardTabs[dashboardsProvider.dashboardTabIndex]
+            .selectedTabColor,
+        unselectedItemColor: Theme.of(context).accentColor,
+        currentIndex: dashboardsProvider.dashboardTabIndex,
+        items: [
+          BottomNavigationBarItem(
+            backgroundColor: Theme.of(context).primaryColor,
+            icon: Icon(Icons.star),
+            title: Text('Favourite'),
```
Comment (JakubKJ) [inline lib/screens/dashboards_screen.dart:38]: I would recommend to keep all strings in one file (example here: https://stackoverflow.com/a/53653788) to easily maintain all of them, prevent defining duplicated strings and to support various languages

**Review by szymon-owczarzak — APPROVED** [review]

File: lib/screens/dashboards_screen.dart (hunk 5 above)
Comment (szymon-owczarzak) [reply lib/screens/dashboards_screen.dart:38]: 👍

File: lib/screens/login_screen.dart [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+                        },
+                        child: Container(
+                          height: 60,
+                          width: double.infinity,
+                          decoration: BoxDecoration(
+                              border: Border.all(color: Colors.white),
+                          ),
+                          child: Center(
+                            child: Text(
+                              "Submit",
```
Comment (szymon-owczarzak) [inline lib/screens/login_screen.dart:44]: All the labels visible to the user should be kept in some translation files so we can implement language switch easily in the future.

File: lib/screens/dashboards_screen.dart (hunk 5 above)
Comment (szymon-owczarzak) [reply lib/screens/dashboards_screen.dart:38]: Please implement this for all Labels. Let the names of the variables describe where they are used if possible.

**Review by szymon-owczarzak — CHANGES_REQUESTED** [review]
Please implement - multi-lang support for future (on UK for now)

**Pull request merged**